src/entrystore/src/seg/resp.rs

```diff
@@ -39,15 +42,45 @@ impl Storage for Seg {
             ExpireTime::Seconds(n) => n,
             _ => 0,
         };
+        let value = match parse_signed_redis(set.value()) {
+            Some(integer) => U64(integer as u64),
+            None => Bytes(set.value()),
+        };
 
         if self
             .data
-            .insert(set.key(), set.value(), None, Duration::from_secs(ttl))
+            .insert(set.key(), value, None, Duration::from_secs(ttl))
             .is_ok()
         {
             Response::simple_string("OK")
         } else {
             Response::error("not stored")
         }
     }
+
+    fn incr(&mut self, incr: &Incr) -> Response {
+        if let Some(mut item) = self.data.get(incr.key()) {
+            match item.value() {
+                seg::Value::Bytes(b) => Response::error("wrong data type"),
+                seg::Value::U64(uint) => {
+                    if let Some(incremented) = (uint as i64).checked_add(1) {
+                        item.wrapping_add(1);
```

Comment:
I don't think this is going to work like you think it does.

Reply:
![image](https://user-images.githubusercontent.com/833575/217624286-d9017979-69f3-4d16-9759-dbe966522a0b.png)
since we already check if the add would overflow if we interpret the u64 as an i64 on line 66, then wrapping add should be safe, right?

Reply:
also this to show what happens w/ negative integers:
![image](https://user-images.githubusercontent.com/833575/217625018-3bafd911-c553-437f-a1e6-c5dd5f6e7b00.png)